### DISCUSSION TITLE
Mermaid in docs disappered (`3.0.0`)

### DISCUSSION THREAD

**Discussion by Zaitonn** [facebook/docusaurus#9584] [Q&A] [answered]
i was upgrading my docs from `v2` to `3.0.0`, but i came across some problem

Here is my config. i have set `markdown.mermaid` true and added `'@docusaurus/theme-mermaid'` to `themeConfig.themes`

https://github.com/iPanelDev/ipaneldev.github.io/blob/867813dcfeb2407ccae48f81f3dc7b5ceee2d44e/docusaurus.config.ts#L24-L27

and i'm sure i have updated all docusaurus package to `3.0.0`

![image](https://github.com/facebook/docusaurus/assets/103164490/e2f93af9-a54e-416b-ba04-e0974dc4a669)

file: https://github.com/iPanelDev/ipaneldev.github.io/blob/2.3/docs/guide/composition/README.mdx

as i expected, there would be a mermaid image at https://ipaneldev.github.io/docs/guide/composition/

but acturally i got nothing
![image](https://github.com/facebook/docusaurus/assets/103164490/c786c7e8-3e2f-45a4-a5c8-14a16284eb81)

the page is built by github actions and it seems same as my local preview or build 

---

are there any mistakes in my config or md file?
:(

**Comment by slorber** [comment, marked as answer]
I don't see any particular issue with your config

Please create a smaller runnable repro using docusaurus.new, this makes it easier to help you

You can try deleting node modules, fully regenerating your lockfile and also running `docusaurus clear`

**Reply by Zaitonn** [reply]
i forked the original repo and deleted all irrelevant files:
https://github.com/Zaitonn/ipaneldev.github.io

but it seems not to change

![image](https://github.com/facebook/docusaurus/assets/103164490/61464594-68b5-41a7-84f6-aaf4e9ad1f53)

![image](https://github.com/facebook/docusaurus/assets/103164490/b3bc64c0-df98-4a13-89ea-979b20c0307b)

>You can try deleting node modules, fully regenerating your lockfile and also running docusaurus clear

same, too :((((((

**Reply by Zaitonn** [reply]
>Please create a smaller runnable repro using docusaurus.new, this makes it easier to help you

i'll try it tomorrow

**Reply by Zaitonn** [reply]
oh it's my fault

i put the theme config in a wrong place :((((((

sorry for taking up your time :(((((( orz

```diff
diff --git a/docusaurus.config.ts b/docusaurus.config.ts
index ea77521..9630f3c 100644
--- a/docusaurus.config.ts
+++ b/docusaurus.config.ts
@@ -110,18 +110,17 @@ const config: Config = {
       darkTheme: prismThemes.dracula,
       additionalLanguages: ['json', 'typescript', 'powershell', 'regex'],
     },
-
-    themes: [
-      '@docusaurus/theme-mermaid',
-      [
-        require.resolve('@easyops-cn/docusaurus-search-local'),
-        {
-          hashed: true,
-          language: ['en', 'zh'],
-        },
-      ],
-    ],
   } satisfies Preset.ThemeConfig,
+  themes: [
+    '@docusaurus/theme-mermaid',
+    [
+      require.resolve('@easyops-cn/docusaurus-search-local'),
+      {
+        hashed: true,
+        language: ['en', 'zh'],
+      },
+    ],
+  ],
 };

 export default config;
```

**Reply by slorber** [reply]
np 😄 that's why I ask for a smaller repro: users usually figure the problem on their own
